inc/azure_c_shared_utility/refcount.h

```diff
@@ -37,28 +37,75 @@ struct C2(C2(REFCOUNT_, type), _TAG)
 C2(REFCOUNT_, type)
 
 #define REFCOUNT_TYPE_DECLARE_CREATE(type) C2(REFCOUNT_SHORT_TYPE(type), _Create)
+#define REFCOUNT_TYPE_DECLARE_CREATE_WITH_EXTRA_SIZE(type) C2(REFCOUNT_SHORT_TYPE(type), _Create_With_Extra_Size)
 #define REFCOUNT_TYPE_CREATE(type) C2(REFCOUNT_SHORT_TYPE(type), _Create)()
+#define REFCOUNT_TYPE_CREATE_WITH_EXTRA_SIZE(type, size) C2(REFCOUNT_SHORT_TYPE(type), _Create_With_Extra_Size)(size)
+#define REFCOUNT_TYPE_DECLARE_DESTROY(type) C2(REFCOUNT_SHORT_TYPE(type), _Destroy)
+#define REFCOUNT_TYPE_DESTROY(type, var) C2(REFCOUNT_SHORT_TYPE(type), _Destroy)(var)
 
 /*this introduces a new refcount'd type based on another type */
 /*and an initializer for that new type that also sets the ref count to 1. The type must not have a flexible array*/
 /*the newly allocated memory shall be free'd by free()*/
 /*and the ref counting is handled internally by the type in the _Create/ _Clone /_Destroy functions */
```

Comment:
>_Create [](start = 64, length = 7)

and _Create_With_Extra_Size

Reply:
Coooorect, added.

---
In reply to: [229833606](https://github.com/Azure/azure-c-shared-utility/pull/221#discussion_r229833606) [](ancestors = 229833606)